multi-node/aws/pkg/cluster/cluster.go

```diff
@@ -309,51 +309,62 @@ func (c *Cluster) validateKeyPair(ec2Svc ec2Service) error {
 type r53Service interface {
 	ListHostedZonesByName(*route53.ListHostedZonesByNameInput) (*route53.ListHostedZonesByNameOutput, error)
 	ListResourceRecordSets(*route53.ListResourceRecordSetsInput) (*route53.ListResourceRecordSetsOutput, error)
+	GetHostedZone(*route53.GetHostedZoneInput) (*route53.GetHostedZoneOutput, error)
 }
 
 func (c *Cluster) validateDNSConfig(r53 r53Service) error {
 	if !c.CreateRecordSet {
 		return nil
 	}
 
-	listHostedZoneInput := route53.ListHostedZonesByNameInput{
-		DNSName: aws.String(c.HostedZone),
-	}
+	if c.HostedZoneID == "" {
+		//TODO(colhom): When HostedZone parameter is gone, this block can be removed
+		//Config will gaurantee that HostedZoneID is set from the get-go
+		listHostedZoneInput := route53.ListHostedZonesByNameInput{
+			DNSName: aws.String(c.HostedZone),
+		}
 
-	zonesResp, err := r53.ListHostedZonesByName(&listHostedZoneInput)
-	if err != nil {
-		return fmt.Errorf("Error validating HostedZone: %s", err)
-	}
+		zonesResp, err := r53.ListHostedZonesByName(&listHostedZoneInput)
+		if err != nil {
+			return fmt.Errorf("Error validating HostedZone: %s", err)
+		}
 
-	zones := zonesResp.HostedZones
+		zones := zonesResp.HostedZones
 
-	if len(zones) == 0 {
-		return fmt.Errorf("hosted zone %s does not exist", c.HostedZone)
-	}
+		if len(zones) == 0 {
+			return fmt.Errorf("hosted zone %s does not exist", c.HostedZone)
+		}
 
-	var matchingZone *route53.HostedZone = nil
-	for _, zone := range zones {
-		if aws.StringValue(zone.Name) == c.HostedZone {
-			//TODO:(chom) for v1, we will require HostedZoneID be
-			//set, as HostedZone will be going away
-			if c.HostedZoneID == "" || aws.StringValue(zone.Id) == c.HostedZoneID {
+		var matchingZone *route53.HostedZone = nil
+		for _, zone := range zones {
+			if aws.StringValue(zone.Name) == c.HostedZone {
 				matchingZone = zone
```

Comment:
Is it safe to assume that we can use the first matching zone, or should we fail on multiple matches?